crypto/scrypt.cpp:39:20: error: redefinition of 'be32enc' on FreeBSD

Building on FreeBSD gives the following error:

```% gmake
Making all in src
gmake[1]: Entering directory '/usr/home/clint/PIVX/src'
gmake[2]: Entering directory '/usr/home/clint/PIVX/src'
  CXX      libbitcoinconsensus_la-allocators.lo
  CXX      primitives/libbitcoinconsensus_la-transaction.lo
  CXX      crypto/libbitcoinconsensus_la-hmac_sha512.lo
  CXX      crypto/libbitcoinconsensus_la-scrypt.lo
crypto/scrypt.cpp:39:20: error: redefinition of 'be32enc'
static inline void be32enc(void *pp, uint32_t x)
                   ^
/usr/include/sys/endian.h:157:1: note: previous definition is here
be32enc(void *pp, uint32_t u)
^
1 error generated.
gmake[2]: *** [Makefile:6103: crypto/libbitcoinconsensus_la-scrypt.lo] Error 1
gmake[2]: Leaving directory '/usr/home/clint/PIVX/src'
gmake[1]: *** [Makefile:7009: all-recursive] Error 1
gmake[1]: Leaving directory '/usr/home/clint/PIVX/src'
gmake: *** [Makefile:592: all-recursive] Error 1
```

--- a/src/crypto/scrypt.cpp
+++ b/src/crypto/scrypt.cpp
@@ -36,6 +36,7 @@
 #include <string.h>
 #include <stdint.h>
 
+#ifndef __FreeBSD__
 static inline void be32enc(void *pp, uint32_t x)
 {
     uint8_t *p = (uint8_t *)pp;
@@ -44,6 +45,7 @@ static inline void be32enc(void *pp, uint32_t x)
     p[1] = (x >> 16) & 0xff;
     p[0] = (x >> 24) & 0xff;
 }
+#endif
 
 typedef struct HMAC_SHA256Context {
     SHA256_CTX ictx;

